src/nvim/api/buffer.c

```diff
@@ -703,6 +708,360 @@ ArrayOf(Integer, 2) nvim_buf_get_mark(Buffer buffer, String name, Error *err)
+  ADD(rv, INTEGER_OBJ((Integer)extmark->col));
+  return rv;
+}
+
+/// Returns an ordered list of mark info
+///
+/// @param buffer The buffer handle
+/// @param[out] err Details of an error that may have occurred
+/// @return [(mark_id, row, col), ...]
+ArrayOf(Object) nvim_buf_get_mark_ids(Buffer buffer,
+                                      Integer namespace,
+                                      Error *err)
+    FUNC_API_SINCE(1)
+{
+  Array rv = ARRAY_DICT_INIT;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if (!buf) {
+    return rv;
+  }
+
+  uint64_t ns = (uint64_t)namespace;
+  Array mark = ARRAY_DICT_INIT;
+  FOR_ALL_EXTMARKS(buf, ns, 0, 0, -1, -1, {
+    if (extmark->ns_id == ns) {
+      mark.size = 0;
+      mark.capacity = 0;
+      mark.items = 0;
+      ADD(mark, INTEGER_OBJ((Integer)extmark->mark_id));
+      ADD(mark, INTEGER_OBJ(extmark->line->lnum));
+      ADD(mark, INTEGER_OBJ(extmark->col));
+      ADD(rv, ARRAY_OBJ(mark));
+    }
+  })
+  return rv;
+}
+
+/// Returns mark info of the mark following the given index
+/// If there are no following marks returns an empty list
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with extmark_ns_create
+/// @param id The mark's id
+/// @param[out] err Details of an error that may have occurred
+/// @return (id, row, col)
+ArrayOf(Integer, 3) nvim_buf_get_next_mark(Buffer buffer,
+                                           Integer namespace,
+                                           Object id,
+                                           Error *err)
+    FUNC_API_SINCE(1)
+{
+  Array rv = ARRAY_DICT_INIT;
+  if (!ns_initialized((uint64_t)namespace)) {
+    api_set_error(err, Validation, _("Invalid mark namespace"));
+    return rv;
+  }
+
+  linenr_T lnum;
+  colnr_T col;
+  bool _err = false;
+  // extremity=false
+  SET_EXTMARK_INDEX_FROM_OBJ(id, lnum, col, _err, false);
+  if (!_err) {
+    return rv;
+  }
+
+  bool match = false;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+  ExtendedMark *next = extmark_next(buf,
+                                    (uint64_t)namespace,
+                                    lnum,
+                                    col,
+                                    match);
+  if (next) {
+    ADD(rv, INTEGER_OBJ((Integer)next->mark_id));
+    ADD(rv, INTEGER_OBJ(next->line->lnum));
+    ADD(rv, INTEGER_OBJ(next->col));
+  }
+  return rv;
+}
+
+/// Returns mark info inbetween and including the bounds
+/// If there are no following marks returns empty list
+///
+/// @param buffer The buffer handle
+/// @param namespace a identifier returned previously with extmark_ns_create
+/// @param lower (row, col) or mark_id of lower bound
+/// @param uower (row, col) or mark_id of upper bound
+/// @param[out] err Details of an error that may have occurred
+/// @return [(mark_id, row, col), ...]
+ArrayOf(Object) nvim_buf_get_next_markrange(Buffer buffer,
```

Comment:
As a precedent, we decided to similarly remove superfluous aliases for get/set buffer line functionality. It might dealing with single marks a bit more inconvenient, but same as for buffer lines it serves a point to nudge API users towards using more efficient range API. Wrappers for convenience can be done on client/vimscript side, we prefer to not have unnecessary core API functions.